### PR TITLE
Docs - add example conda-dev-env

### PR DESCRIPTION
**Pull request by dazza-codes**
Split out of #3398 to be rebased later and adapted to provide a small snippet for dev-getting-started.

The intention is to provide a small one-stop-shop for a copy-paste snippet that works, with no doubts or questions about how to use the files involved.  There is great value in DRY consistent documentation, so it could be removed for the sake of doc-maintenance and consistency.  But, it currently places a burden on the reader to jump around to a few places to sift through the volumes to find a few kernels about how to get off the ground.

### CODE REVIEW

File: docs/source/develop.rst [hunk 1]
```diff
@@ -22,6 +22,26 @@ and install it from source::
...
 
+Using conda, for example::
+
+   git clone git@github.com:{your-fork}/distributed.git
+   cd distributed
+   conda create -y -n distributed python=3.6
+   conda activate distributed
+   pip install -U -r requirements.txt
+   pip install -U -r dev-requirements.txt
+   pip install -e .
```
Comment (quasiben) [inline docs/source/develop.rst]: I think the recommended way of using pip now is with `python -m pip ...`. but I can't find where are read that.

Comment (TomAugspurger) [reply docs/source/develop.rst]: https://snarky.ca/why-you-should-use-python-m-pip/

Comment (dazza-codes) [reply docs/source/develop.rst]: Done

File: docs/source/develop.rst [hunk 2]
```diff
@@ -22,6 +22,26 @@ and install it from source::
...
+   python -m pip install -U -r dev-requirements.txt
+   python -m pip install -e .
+
+To keep a fork in sync with the upstream source::
+
+   cd distributed
+   git checkout master
+   git remote add upstream git@github.com:dask/distributed.git
+   git remote -v
+   git fetch -ap upstream
```
Comment (quasiben) [inline docs/source/develop.rst]: One last comment here, the `-p` may unintentionally remove tags the user would like to keep.  I don't think this is the common case, but if we are going to be recommending git commands perhaps we should avoid prunng?

Comment (dazza-codes) [reply docs/source/develop.rst]: OK, better safe than sorry.  Done.

**Comment by quasiben** [comment]
Thanks @dazza-codes for the contribution -- merging in!

**Pull request merged**